src/amo/reducers/users.js

```diff
@@ -317,16 +390,29 @@ const reducer = (
 
       return {
         ...state,
-        ...addUserToState({ state, user }),
+        ...addUserToState({ notifications: null, state, user }),
         currentUserID: user.id,
       };
     }
     case LOAD_USER_ACCOUNT: {
-      const { user } = action.payload;
+      const { notifications, user } = action.payload;
+
+      return {
+        ...state,
+        ...addUserToState({ notifications, state, user }),
+      };
+    }
+    case LOAD_USER_NOTIFICATIONS: {
+      const { notifications, username } = action.payload;
+      const user = getUserByUsername(state, username);
+
+      if (!user) {
```

Comment:
When would this happen, and if it does happen should that be an error?

Reply:
Redux `reducer()` function should be pure and always return a state, which is why I'd like not to throw errors here.

I believe Flow encouraged me to ensure we have a user. In real life, there is no way to load notifications without the user being loaded first.